crates/settlement-clients/ethereum/src/conversion.rs

```diff
@@ -1,12 +1,325 @@
+    ], U256::from_str_radix("00112233445566778899AABBCCDDEEFF00112233445566778899AABBCCDDEEFF", 16).unwrap())]
+    #[case::minimum(&[0; 32], U256::ZERO)]
+    #[case::maximum(&[0xFF; 32], U256::MAX)]
+    #[case::short(&[0xFF; 16], U256::from_be_slice(&[0xFF; 16]))]
+    #[case::empty(&[], U256::ZERO)]
+    fn slice_u8_to_u256_works(#[case] slice: &[u8], #[case] expected: U256) {
+        match slice_u8_to_u256(slice) {
+            Ok(response) => {
+                assert_eq!(response, expected);
+            }
+            Err(e) => {
+                panic!("{}", e);
+            }
+        }
+    }
+
+    #[rstest]
+    #[case::over(&[0xFF; 33])]
+    fn slice_u8_to_u256_panics(#[case] slice: &[u8]) {
+        let result: Result<alloy::primitives::Uint<256, 4>, color_eyre::eyre::Error> = slice_u8_to_u256(slice);
+        match result {
+            Ok(_) => {
+                panic!("{}", "Should not have passed");
+            }
+            Err(report) => {
+                assert_eq!(report.to_string(), "could not convert &[u8] to U256")
+            }
+        }
+    }
+
+    #[rstest]
+    #[case::empty(&[], vec![])]
+    #[case::single(
+        &[[1; 32]],
+        vec![U256::from_be_slice(&[1; 32])]
+    )]
+    #[case::multiple(
+        &[
+            [1; 32],
+            [2; 32],
+            [3; 32],
+        ],
+        vec![
+            U256::from_be_slice(&[1; 32]),
+            U256::from_be_slice(&[2; 32]),
+            U256::from_be_slice(&[3; 32]),
+        ]
+    )]
+    #[case::mixed(
+        &[
+            [0xFF; 32],
+            [0x00; 32],
+            [0x12, 0x34, 0x56, 0x78, 0x9A, 0xBC, 0xDE, 0xF0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        ],
+        vec![
+            U256::MAX,
+            U256::ZERO,
+            U256::from_be_slice(&[0x12, 0x34, 0x56, 0x78, 0x9A, 0xBC, 0xDE, 0xF0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]),
+        ]
+    )]
+    fn slice_slice_u8_to_vec_u256_works(#[case] slices: &[[u8; 32]], #[case] expected: Vec<U256>) {
+        match slice_slice_u8_to_vec_u256(slices) {
+            Ok(response) => {
+                assert_eq!(response, expected);
+            }
+            Err(e) => {
+                panic!("{}", e);
+            }
+        }
+    }
+
+    #[rstest]
+    #[case::empty(&[], "0".repeat(64))]
+    #[case::typical(&[0xFF,0xFF,0xFF,0xFF], format!("{}{}", "ff".repeat(4), "0".repeat(56)))]
+    #[case::big(&[0xFF; 32], format!("{}", "ff".repeat(32)))]
+    fn to_hex_string_works(#[case] slice: &[u8], #[case] expected: String) {
+        let result = to_padded_hex(slice);
+        assert_eq!(result, expected);
+        assert!(expected.len() == 64);
+    }
+
+    #[rstest]
+    #[should_panic(expected = "Slice length must not exceed 32")]
+    #[case::exceeding(&[0xFF; 40], format!("{}", "ff".repeat(32)))]
+    fn to_hex_string_panics(#[case] slice: &[u8], #[case] expected: String) {
```

Comment:
same as above, i think we can combine test cases

Reply:
We can, but how do you think the naming of the function be, right now there is a clear decoupling between failing and passing cases naming separating two types of tests.

WDYT ?

Reply:
same as above

Reply:
Alright, Implemented